### ISSUE TITLE
Fixes to documentation

### ISSUE THREAD

**Issue by estevegf** [raml-org/raml-spec#729]
Hi, 

After reading the docs I've found some issues (please do apologize me if any of them is a misunderstanding):

- "RAML does not define validation when a header declaration specifies any of the following types for the value of the header: an object type, **a union of non-scalar types, or an array type if the underlying type of the array is an object type**, **array type, or a union of non-scalar types**."  
  - a union of non-scalar types is repeated
  - an array type if the underlying type of the array is an object type seems to be mutually exclusive to array type assertion
- types:
``` yaml
Email:
    type: string
    minLength: 2
    maxLength: 6
    pattern: **^note\d+$**
```
probably you're pasting the pattern from an example in another section above in the docs.

-  "sets the value of resourcePathName and resourcePath parameters to "/bom/{itemId}" and "bom", respectively." I guess this statement should be reversed.

I wait for your news.

Thanks

**Comment by stock-dd** [comment]
16feb and no reaction? Wow looks like a great community. About ready to take off to Swagger.

**Comment by jstoiko** [comment]
Thanks for this @estevegf. Added some your points to be released as part of the v1.0.1 patch.

Regarding:
- your 1st point: I do not see any issue with that sentence. It should be read as follows:
```
RAML does not define validation when a header declaration specifies any of the following types for the value of the header: 
- an object type, 
- a union of non-scalar types, or 
- an array type if the underlying type of the array is 
    - an object type, 
    - array type, or 
    - a union of non-scalar types.
```
- your 2nd point: good catch. See b76d1d0.
- your 3rd point: good catch. See 8a45c85.

**Comment by estevegf** [comment]
Good Work!

Keep working 💪
